src/control/cmd/daos_agent/security_rpc_test.go

```diff
@@ -317,19 +317,29 @@ func TestAgent_SecurityRPC_getCredential(t *testing.T) {
 			conn, cleanup := setupTestUnixConn(t)
 			defer cleanup()
 
-			mod := NewSecurityModule(log, tc.secCfg)
-			mod.signCredential = func() credSignerFn {
-				var idx int
-				return func(_ context.Context, req *auth.CredentialRequest) (*auth.Credential, error) {
-					defer func() {
-						if idx < len(tc.responses)-1 {
-							idx++
-						}
-					}()
-					t.Logf("returning response %d: %+v", idx, tc.responses[idx])
-					return tc.responses[idx].cred, tc.responses[idx].err
+			CredentialRequestGetSignedTesting := func(ctx context.Context, log logging.Logger, req auth.CredentialRequest) (*auth.Credential, error) {
+				r, ok := req.(*auth.CredentialRequestUnix)
+				if !ok {
+					return nil, errors.New("Testing fn used with non-unix credential request")
 				}
-			}()
+				testing_fn := func() auth.GetSignedCredentialInternalFn {
+					var idx int
+					return func(_ context.Context, _ *auth.CredentialRequestUnix) (*auth.Credential, error) {
+						defer func() {
+							if idx < len(tc.responses)-1 {
+								idx++
+							}
+						}()
+						t.Logf("returning response %d: %+v", idx, tc.responses[idx])
+						return tc.responses[idx].cred, tc.responses[idx].err
+					}
+				}()
+				r.GetSignedCredentialInternal = testing_fn
+				return r.GetSignedCredential(log, ctx)
+			}
```

Comment:
This feels a bit over-complex for a mock function. It seems like testing_fn is what you're really trying to inject here. Worth re-thinking the design approach here I think. Best to design the code to be easy to test.